rbac/management/workspace/service.py

```diff
@@ -78,6 +87,30 @@ def destroy(self, instance: Workspace) -> None:
         dual_write_handler.replicate_deleted_workspace()
         instance.delete()
 
+    def _enforce_hierarchy_depth(self, target_parent_id: uuid.UUID, tenant: Tenant) -> None:
+        """Enforce hierarchy depth limits on workspaces."""
+        if self._exceeds_depth_limit(target_parent_id, tenant):
+            message = f"Workspaces may only nest {settings.WORKSPACE_HIERARCHY_DEPTH_LIMIT} levels deep."
+            error = {"workspace": [message]}
+            raise serializers.ValidationError(error)
+        if self._violates_peer_restrictions(target_parent_id, tenant):
+            message = "Sub-workspaces may only be created under the default workspace."
+            error = {"workspace": [message]}
+            raise serializers.ValidationError(error)
+
     def _parent_id_attr_update(self, attr: str, value: str, instance: Workspace) -> bool:
         """Determine if the attribute being updated is parent_id."""
         return attr == "parent_id" and instance.parent_id != value
+
+    def _violates_peer_restrictions(self, target_parent_id: uuid.UUID, tenant: Tenant) -> bool:
+        """Determine if peer restrictions are violated."""
+        target_root_workspace = Workspace.objects.root(tenant=tenant)
+        if settings.WORKSPACE_RESTRICT_DEFAULT_PEERS and str(target_root_workspace.id) == str(target_parent_id):
```

Comment:
what about ungrouped workspaces? it is expected to allow to create sub-workspaces?

Reply:
@petracihalova it is not. Only hosts could be in ungrouped. Do we allow to created sub workspaces ?

Reply:
[RHCLOUD-40242](https://issues.redhat.com/browse/RHCLOUD-40242)